src/main/java/org/mockito/internal/matchers/CapturingMatcher.java

```diff
@@ -26,16 +28,17 @@ public String toString() {
     }
 
     public T getLastValue() {
-        if (arguments.isEmpty()) {
-            throw noArgumentValueWasCaptured();
-        }
-
-        return (T) arguments.getLast();
+        synchronized (arguments) {
```

Comment:
I don't understand how you prevent a race condition if you only synchronize the read/getLastValue() operation but not not the write/captureFrom() operation. Again the synchronizedList uses it's own private lock object to make single operations atomar.
It is still possible that one Thread add a new argument while an other Thread is in `getLastValue()` after `arguments.size(..)` and before  `arguments.get(..)`.